src/tiledb/cloud/file/utils.py

```diff
@@ -10,6 +13,31 @@
 from tiledb.cloud.rest_api import models
 
 
+def sanitize_filename(fname: str) -> str:
+    """
+    Sanitizes a filename by removing invalid characters.
+
+    :param fname: A filename to sanitize
+    :return str: The sanitized string
+    """
+    name, suffix = os.path.splitext(fname)
+    name = re.sub(r"[^\w\s-]", "", name)
+    name = re.sub(r"[-_\s]+", "_", name).strip("-_")
+    return name + suffix
+
+
+def basename_match(file_uri: str, pattern: Optional[str] = None) -> bool:
+    """
+    Checks if the basename of a given file uri matches
+    the given UNIX shell style pattern.
+
+    :param file_uri: A file URI.
+    :param pattern: A UNIX shell style pattern, defaults to None
+    :return bool: Pattern matches the file basename or not.
+    """
+    return pattern is not None and fnmatch(os.path.basename(file_uri), pattern)
```

Comment:
If the pattern is None, do we want to reject or accept?

Reply:
If the pattern is `None` then it cannot match or we don't care about matching (this is for the method to be more general). 
Also guards against a type error being raised from fnmatch in that case.

Reply:
My thought here is that if we don't care about matching, then we would want to always say yes.

Reply:
We do care about matching, that is why if the `pattern` is `None` we return `False`.
Also semantically, the method's name is `basename_match` so with `None` I would expect not to match